sdk/aqueduct/decorator.py

```diff
@@ -71,17 +73,19 @@ def wrap_spec(
         description:
             The description for this operator.
     """
-    if len(input_artifacts) < 1:
-        raise Exception("Transformation requires at least one input artifact.")
-
     for artifact in input_artifacts:
         if artifact._from_flow_run:
             raise InvalidUserActionException(
                 "Artifact fetched from flow run can not be reused for computation"
             )
 
-    dag = input_artifacts[0]._dag
-    api_client = input_artifacts[0]._api_client
+    if len(input_artifacts) == 0:
+        print("This operator takes no artifact input.")
+        dag = dag_module.__GLOBAL_DAG__
+        api_client = api_client_module.__GLOBAL_API_CLIENT__
```

Comment:
There's another assumption we're sort of breaking here. Previously, all starting operators have to be constructed through the aqueduct client, so the client would have to be instantiated before running operators. Now, a user may expect their decorated function to run even without the client instantiated, and will be confused as to why the client needs to be constructed even if it's not directly used.

I don't think we can avoid this dependency right now, but what if we communicated this better by:
- initialize the global dag to `DAG(metadata=Metadata())` instead of None.
- initialize the global api client without the api_key and server_address. We can add a `configure(api_key, server_address)` method to set these fields up. If we attempt to use the api_client without configuring it, the user will get a very informative error about how they need to set the api_key and server_address first, currently by instantiating the client. Later on, when we have the concept of a `aq.config(...)` global config object, we can configure the api_client directly there.